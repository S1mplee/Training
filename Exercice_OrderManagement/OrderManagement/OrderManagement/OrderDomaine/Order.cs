using ReactiveDomain;
using ReactiveDomain.Messaging;
using System;

namespace TESTDICT
{
    public class Order : EventDrivenStateMachine
    {
        private string Action;
        private string Asset;
        private decimal Price;
        private int Quantite;
        public Order()
        {
            Register<OrderCreated>(evt =>
            {
                this.Id = evt.OrderId;
                this.Action = evt.Action;
                this.Asset = evt.Asset;
                this.Price = evt.Price;
                this.Quantite = evt.Quantite;
            });
        }
        public Order(Guid id,string action,string asset,decimal price,int qts) : this()
        {
            if (id == null || string.IsNullOrEmpty(action) || string.IsNullOrEmpty(asset) || price <= 0 || qts <= 0) throw new ArgumentException("Invalid Inputs !");
            Raise(new OrderCreated(id, action, asset, price, qts));
        }

     class MySecretEvent : Message { }

    }
}
