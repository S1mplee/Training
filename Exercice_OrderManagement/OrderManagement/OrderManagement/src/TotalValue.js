import React from 'react';

class TotalValue extends React.Component {
    constructor(props) {
        super(props);
    }


    render() {
        return (
            <div className="col">
                <div className="card border-danger mb-3 m-2" style={{ width: '18rem' }}>
                    <div className="card-header">Total Value :</div>
                    <div className="card-body text-danger">
                        <h5 className="card-title">{this.props.Total} $</h5>
                    </div>
                </div>
            </div>
            );
    }

}

export default TotalValue;
