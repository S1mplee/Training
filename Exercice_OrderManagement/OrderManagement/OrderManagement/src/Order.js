import React from 'react';
import dotnetify from 'dotnetify';
import TotalValue from './TotalValue';
import Asset from './Assets';

export default class Order extends React.Component {
    constructor(props) {
        super(props);
        this.vm = dotnetify.react.connect("ViewModel", this);
        this.state = {
            d: "", list: [], textMessage: "", TextBoxValue: '',
            SimpleDropDownValue: '',
            SimpleDropDownOptions: [],
            DropDownValue: '',
            DropDownOptions: [],
            SimpleDropDownValue2: '',
            SimpleDropDownOptions2: [],
            DropDownValue2: '',
            DropDownOptions2: [],
            Price: "",
            Qts: "",
            Assets: [],
            dict: {},
            Total: "",
            message1: "",
            message2: "",
            message3: ""

        };
        this.dispatchState = state => this.vm.$dispatch(state);

    }

    componentWillUnmount() {
        this.vm.$destroy();
    }

    Clickme() {

    }
    
   
    render() {

        return (
          
            <div className="container contact-form">
           
                <div className="container">
                    <div className="row">
                      
                        <div className="col">
                            <TotalValue Total={this.state.Total} />
    </div>
                        <div className="col">
                            <div className="card border-primary  mb-3 m-2" style={{ width: '20rem' }}>
                                <div className="card-header">Assets :</div>
                                <Asset dict={this.state.dict} />
                            </div>
    </div>
                    </div>
                </div>
               
                <form className="dd" method="post">
                    <h3>Send Order</h3>
                    <label style={{ color:'red' }}> {this.state.message3} </label>

                    <div className="row">
                        <div className="col-md-6">
                            <div className="form-group">
                                <label for="exampleFormControlTextarea6" style={{ color: 'green' }} className="m-2">Action : </label>

                                <select
                                    className="form-control"
                                    value={this.state.SimpleDropDownValue}
                                    onChange={e => this.dispatchState({ SimpleDropDownValue: e.target.value })}
                                >
                                    <option value="" disabled>
                                        Choose...
                  </option>
                                    {this.state.SimpleDropDownOptions.map((text, idx) => (
                                        <option key={idx} value={text}>
                                            {text}
                                        </option>
                                    ))}
                                </select>
                                <b>{this.state.SimpleDropDownResult}</b>
                            </div>
                            <div className="form-group">
                                <label for="exampleFormControlTextarea6" style={{ color: 'green' }}>Price : </label>
                                <input  onBlur={_ => this.dispatchState({ Price: this.state.Price })}
                                    onChange={e => this.setState({ Price: e.target.value })} value={this.state.Price} className="form-control" placeholder="Price" />
                            </div>
                            <label className="m-2" style={{ color: 'red' }}>{this.state.message1}</label>
                            <div className="form-group">
                                <label for="exampleFormControlTextarea6" style={{ color: 'green' }}>Asset : </label>

                                <select
                                    className="form-control"
                                    value={this.state.SimpleDropDownValue2}
                                    onChange={e => this.dispatchState({ SimpleDropDownValue2: e.target.value })}
                                >
                                    <option value="" disabled>
                                        Choose...
                  </option>
                                    {this.state.SimpleDropDownOptions2.map((text, idx) => (
                                        <option key={idx} value={text}>
                                            {text}
                                        </option>
                                    ))}
                                </select>
                                <b>{this.state.SimpleDropDownResult2}</b>
                            </div>
                            <div className="form-group">
                                <label for="exampleFormControlTextarea6" style={{ color: 'green' }}>Quantite : </label>

                                <input onBlur={_ => this.dispatchState({ Qts: this.state.Qts })}
                                    onChange={e => this.setState({ Qts: e.target.value })} value={this.state.Qts} className="form-control" placeholder="Quantite" />
                                <label className="m-2" style={{ color: 'red' }}>{this.state.message2}</label>
                            </div>
                            <div className="form-group">
                                <input type="button" onClick={_ => this.dispatchState({ ButtonClicked: true })} name="btnSubmit" className="btnContact" value="Send Order" />
                            </div>
                        </div>
                        <div className="col-md-6">
                            <div className="form-group">
                                <div className="form-group shadow-textarea">
                                    <label for="exampleFormControlTextarea6" style={{ color: 'green' }}>History : </label>
                                    <textarea className="form-control z-depth-1" id="exampleFormControlTextarea6" rows="3" placeholder="History." value={this.state.history}></textarea>
                                </div>
                            </div>
                        </div>
                    </div>
                </form>
            </div>
        );
    }
    }


