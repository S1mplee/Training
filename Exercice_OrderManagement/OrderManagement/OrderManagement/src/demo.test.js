import React from 'react';
import Order from './Order';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import TotalValue from './TotalValue';
import Asset from './Assets';

test(" snapshot test", () => {
    const tree = renderer
        .create(<Order />)
        .toJSON();
    expect(tree).toMatchSnapshot();

});

test('Order : Testing Container Render', () => {
    const component = shallow(<Order />);
    const wrapper = component.find('.container');
    expect(wrapper.length).toBeGreaterThan(0);
 
});

test('Order : Testing Form Render', () => {
    const component = shallow(<Order />);
    const wrapper = component.find('.dd');
    expect(wrapper.length).toBeGreaterThan(0);

});

test('TotalValue : Testing Column Render', () => {
    const component = shallow(<TotalValue />);
    const wrapper = component.find('.col');
    expect(wrapper.length).toBeGreaterThan(0);

});

test('TotalValue : Testing Total prop Render', () => {
    const tree = renderer
        .create(<TotalValue Total='900'/>)
        .toJSON();
    expect(tree).toMatchSnapshot();

    const wrapper = mount(<TotalValue Total="900" />);
    const res = wrapper.find('.card-title');

    expect(res.text()).toBe("900 $");

});

test('TotalValue : Testing Card Render', () => {
    const component = shallow(<TotalValue />);
    const wrapper = component.find('.card-body');
    expect(wrapper.length).toBeGreaterThan(0);

});

test('TotalValue : Testing dict prop Render', () => {
    var dict1 = {};
    dict1["Asset1"] = { Quantite: "5", price: "100" };

    const tree = renderer
        .create(<Asset dict={dict1} />)
        .toJSON();
    expect(tree).toMatchSnapshot();

    const wrapper = mount(<Asset dict={dict1} />);

    const res = wrapper.find('li');

    expect(res.text()).toBe("Asset1 Quantite : 5 Price : 100");


});





