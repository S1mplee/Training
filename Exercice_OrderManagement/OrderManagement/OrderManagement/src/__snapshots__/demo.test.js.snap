// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` snapshot test 1`] = `
<div
  className="container contact-form"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="col"
        >
          <div
            className="card border-danger mb-3 m-2"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <div
              className="card-header"
            >
              Total Value :
            </div>
            <div
              className="card-body text-danger"
            >
              <h5
                className="card-title"
              >
                
                 $
              </h5>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="card border-primary  mb-3 m-2"
          style={
            Object {
              "width": "20rem",
            }
          }
        >
          <div
            className="card-header"
          >
            Assets :
          </div>
          <div
            className="card-body text-primary "
          >
            <ul />
          </div>
        </div>
      </div>
    </div>
  </div>
  <form
    className="dd"
    method="post"
  >
    <h3>
      Send Order
    </h3>
    <label
      style={
        Object {
          "color": "red",
        }
      }
    >
       
      
       
    </label>
    <div
      className="row"
    >
      <div
        className="col-md-6"
      >
        <div
          className="form-group"
        >
          <label
            className="m-2"
            for="exampleFormControlTextarea6"
            style={
              Object {
                "color": "green",
              }
            }
          >
            Action : 
          </label>
          <select
            className="form-control"
            onChange={[Function]}
            value=""
          >
            <option
              disabled={true}
              value=""
            >
              Choose...
            </option>
          </select>
          <b />
        </div>
        <div
          className="form-group"
        >
          <label
            for="exampleFormControlTextarea6"
            style={
              Object {
                "color": "green",
              }
            }
          >
            Price : 
          </label>
          <input
            className="form-control"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Price"
            value=""
          />
        </div>
        <label
          className="m-2"
          style={
            Object {
              "color": "red",
            }
          }
        >
          
        </label>
        <div
          className="form-group"
        >
          <label
            for="exampleFormControlTextarea6"
            style={
              Object {
                "color": "green",
              }
            }
          >
            Asset : 
          </label>
          <select
            className="form-control"
            onChange={[Function]}
            value=""
          >
            <option
              disabled={true}
              value=""
            >
              Choose...
            </option>
          </select>
          <b />
        </div>
        <div
          className="form-group"
        >
          <label
            for="exampleFormControlTextarea6"
            style={
              Object {
                "color": "green",
              }
            }
          >
            Quantite : 
          </label>
          <input
            className="form-control"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Quantite"
            value=""
          />
          <label
            className="m-2"
            style={
              Object {
                "color": "red",
              }
            }
          >
            
          </label>
        </div>
        <div
          className="form-group"
        >
          <input
            className="btnContact"
            name="btnSubmit"
            onClick={[Function]}
            type="button"
            value="Send Order"
          />
        </div>
      </div>
      <div
        className="col-md-6"
      >
        <div
          className="form-group"
        >
          <div
            className="form-group shadow-textarea"
          >
            <label
              for="exampleFormControlTextarea6"
              style={
                Object {
                  "color": "green",
                }
              }
            >
              History : 
            </label>
            <textarea
              className="form-control z-depth-1"
              id="exampleFormControlTextarea6"
              placeholder="History."
              rows="3"
            />
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`TotalValue : Testing Total prop Render 1`] = `
<div
  className="col"
>
  <div
    className="card border-danger mb-3 m-2"
    style={
      Object {
        "width": "18rem",
      }
    }
  >
    <div
      className="card-header"
    >
      Total Value :
    </div>
    <div
      className="card-body text-danger"
    >
      <h5
        className="card-title"
      >
        900
         $
      </h5>
    </div>
  </div>
</div>
`;

exports[`TotalValue : Testing dict prop Render 1`] = `
<div
  className="card-body text-primary "
>
  <ul>
    <li>
      Asset1
       Quantite : 
      5
       Price : 
      100
    </li>
  </ul>
</div>
`;
